source/server/http/admin.cc

```diff
@@ -279,11 +290,42 @@ Http::Code AdminImpl::handlerStats(const std::string&, Buffer::Instance& respons
     all_stats.emplace(gauge->name(), gauge->value());
   }
 
-  for (auto stat : all_stats) {
-    response.add(fmt::format("{}: {}\n", stat.first, stat.second));
+  if (params.size() == 0) {
+    // No Arguments so use the standard
+    for (auto stat : all_stats) {
+      response.add(fmt::format("{}: {}\n", stat.first, stat.second));
+    }
+  } else {
+    std::string format = params.begin()->first;
+    std::string formatvalue = params.begin()->second;
+    if (format == "format" && formatvalue == "json") {
+      rapidjson::Document document;
```

Comment:
nit: Can we move the actual JSON creation into a helper function. We will likely have other output types in the future.

Reply:
will do